pkg/providers/snow/entry.go

```diff
@@ -0,0 +1,74 @@
+package snow
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/networkutils"
+	providerValidator "github.com/aws/eks-anywhere/pkg/providers/validator"
+)
+
+type ConfigManager struct {
+	validator  *Validator
+	defaulters *Defaulters
+}
+
+func NewConfigManager(defaulters *Defaulters, validators *Validator) *ConfigManager {
+	return &ConfigManager{
+		validator:  validators,
+		defaulters: defaulters,
+	}
+}
+
+func (cm *ConfigManager) SetDefaultsAndValidate(ctx context.Context, config *cluster.Config) error {
+	configManager := cluster.NewConfigManager()
+	if err := configManager.Register(cm.snowEntry(ctx)); err != nil {
+		return err
+	}
+
+	if err := configManager.SetDefaults(config); err != nil {
+		return err
+	}
+
+	if err := configManager.Validate(config); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cm *ConfigManager) snowEntry(ctx context.Context) *cluster.ConfigManagerEntry {
+	return &cluster.ConfigManagerEntry{
+		Defaulters: []cluster.Defaulter{
+			func(c *cluster.Config) error {
+				for _, m := range c.SnowMachineConfigs {
+					if err := cm.defaulters.setDefaultSshKeyAndSaveToFile(ctx, m); err != nil {
+						return err
+					}
```

Comment:
I think this embeds too much knowledge for defaulting machine configs. If we macrofy this code to wanting to apply 10 different defaults to a machine config we'll observe 10 different if statements in this inline function, thats not good because its very difficult to test. 

Instead, would we be able to model a `SnowMachineConfigDefaulter` that accepts a `SnowMachineConfig` and knows how to apply all the necessary defaults? Or something to that effect?

Reply:
i'd prefer to have a separate convo w @g-gaston on the configManager entry. this is the format we are using in `pkg/cluster`. if we decide to refactor it we can do it in another pr. Not this one.

Reply:
My 2c: I would be ok with either: what we have here or what @chrisdoherty4 is proposing

I get the fear of this becoming a ball of mud if it keeps growing uncontrolled. However, we currently only have one defaulter of this type and we might not need to add more, so do should we add another entity now or can it wait? I just realized this comment is probably not very helpful, sorry folks 🙂 

As far the config manager is general, it's supposed to just be a "framework" that doesn't necessarily enforce any patterns and it leaves it up to the user to decide what's the best way to use it (multiple defaults in one _defaulter_ closure, multiple closures, nested defaults, etc.

Reply:
We talked about this offline and will address separately. 

To elaborate a little further, consider a desire to validate or default a machine config separate from any `ConfigManager` construct. We can't do that using this approach. We have to either use the `ConfigManager` (which is super odd given I just want to validate something) _or_ duplicate all the rules elsewhere hence we aren't embodying validation in a single re-usable construct. Consequently, I would go as far as to say this is the wrong way to model this stuff.

To your point, Guillermo, I totally agree the `ConfigManager` is a framework and I shared the same thing with Joey. How we construct the actual logic is orthogonal so any existing stuff isn't strictly a 'pattern', its just how its been done for that piece of code for now and if done as is here, should probably change also at some point.